runtime/src/tasks/fifo/tasks-fifo.c

```diff
@@ -1372,26 +1353,38 @@ static void maybe_add_thread(void) {
 // assumes threading_lock has already been acquired!
 static inline
 task_pool_p add_to_task_pool(chpl_fn_p fp,
-                             void* a,
+                             chpl_task_bundle_t* a, size_t a_size,
                              chpl_bool is_executeOn,
+                             chpl_bool countRunningTasks,
                              chpl_task_prvDataImpl_t chpl_data,
                              task_pool_p* p_task_list_head,
                              chpl_bool is_begin_stmt,
                              int lineno, int32_t filename) {
-  task_pool_p ptask =
-    (task_pool_p) chpl_mem_alloc(sizeof(task_pool_t),
-                                        CHPL_RT_MD_TASK_POOL_DESC,
-                                        0, 0);
+
+
+  size_t payload_size;
+  task_pool_p ptask;
+
+  assert(a_size >= sizeof(chpl_task_bundle_t));
+
+  payload_size = a_size - sizeof(chpl_task_bundle_t);
+  ptask = (task_pool_p) chpl_mem_alloc(sizeof(task_pool_t) + payload_size,
+                                       //CHPL_RT_MD_TASK_POOL_DESC,
```

Comment:
*ptask is now both a TASK_POOL_DESC memory and TASK_ARG memory.  We should invent a new CHPL_RT_MD value to reflect this.  The change here makes it different, but no more accurate.